datumaro/plugins/open_images_format.py

```diff
@@ -0,0 +1,155 @@
+
+                set_parents_from_node(child_node, child_category)
+
+        _, root_category = label_categories.find(root_node['LabelName'])
+        set_parents_from_node(root_node, root_category)
+
+    def _load_items(self):
+        items_by_id = {}
+
+        def load_from(path):
+            with self._open_csv_annotation(path) as image_reader:
+                for image_description in image_reader:
+                    image_id = image_description['ImageID']
+                    assert _RE_IMAGE_ID.fullmatch(image_id)
+                    assert image_id not in items_by_id
+
+                    subset = image_description['Subset']
+                    assert subset in {'test', 'train', 'validation'}
+
+                    items_by_id[image_id] = DatasetItem(
+                        id=image_id,
+                        image=osp.join(self._dataset_dir, f'images/{subset}/{image_id}.jpg'),
+                        subset=subset,
+                    )
+
+        # It's preferable to load the combined image description file,
+        # because it contains descriptions for training images without human-annotated labels
+        # (the file specific to the training set doesn't).
+        # However, if it's missing, we'll try loading subset-specific files instead, so that
+        # this exporter can be used on individual subsets of the dataset.
+        try:
+            load_from(_OpenImagesPath.FULL_IMAGE_DESCRIPTION_NAME)
+        except FileNotFoundError:
+            for pattern in _OpenImagesPath.SUBSET_IMAGE_DESCRIPTION_PATTERNS:
+                for path in self._glob_annotations(pattern):
+                    load_from(path)
+
+        self._items.extend(items_by_id.values())
+
+        self._load_labels(items_by_id)
+
+    def _load_labels(self, items_by_id):
+        label_categories = self._categories[AnnotationType.label]
+
+        # TODO: implement reading of machine-annotated labels
+
+        for label_path in self._glob_annotations('*-human-imagelabels.csv'):
+            with self._open_csv_annotation(label_path) as label_reader:
+                for label_description in label_reader:
+                    image_id = label_description['ImageID']
+                    item = items_by_id[image_id]
+
+                    confidence = float(label_description['Confidence'])
+
+                    if confidence > 0.5:
+                        label_name = label_description['LabelName']
+                        label_index, _ = label_categories.find(label_name)
+                        assert label_index is not None
+                        item.annotations.append(Label(label_index))
+
+
+class OpenImagesImporter(Importer):
+    @classmethod
+    def find_sources(cls, path):
+        for pattern in [
+            _OpenImagesPath.FULL_IMAGE_DESCRIPTION_NAME,
+            *_OpenImagesPath.SUBSET_IMAGE_DESCRIPTION_PATTERNS,
+        ]:
+            if glob.glob(osp.join(glob.escape(path), _OpenImagesPath.ANNOTATIONS_DIR, pattern)):
```

Comment:
I don't think calling `glob` in a loop is a good idea. Consider using `cls._find_sources_recursive()` with file name filter.